docs/extendedusage.rst

```diff
@@ -344,6 +351,279 @@ would be ``/foo/bar/{id}``, and the absolute URI path would be
 Check out :doc:`api` for full definition of what is available for a ``resource`` object, and its associated attributes and objects.
 
 
+.. _nonramlparsing:
+
+Non-RAML Parsing
+----------------
+
+JSON
+^^^^
+
+For ``json`` filetypes, ``ramlfications`` will also parse ``$ref`` keywords and bring in the referenced objects according to both `Draft 3`_ and `Draft 4`_ JSON Schema definition.
+
+The following ``$ref`` values are supported:
+
+* :ref:`internalfrag`
+* :ref:`localfile`
+* :ref:`remotefile`
+
+.. _internalfrag:
+
+Internal fragments
+~~~~~~~~~~~~~~~~~~
+
+RAML File:
+
+.. code-block:: yaml
+
+    #%RAML 0.8
+    title: Sample API Demo - JSON Includes
+    version: v1
+    schemas:
+        - json: !include includes/ref_internal_fragment.json
+    baseUri: http://json.example.com
+    /foo:
+      displayName: foo resource
+
+``includes/ref_internal_fragment.json`` file:
+
+.. code-block:: json
+
+    {
+      "references": {
+        "internal": "yes"
+      },
+      "name": "foo",
+      "is_this_internal?": [{"$ref": "#/references/internal"}]
+    }
+
+
+``ramlfications`` would produce the following:
+
+.. code-block:: pycon
+
+    >>> RAML_FILE = "api.raml"
+    >>> api = parse(RAML_FILE)
+    >>> api.schemas
+    [{'json': {u'is_this_internal?': [u'yes'],
+    u'name': u'foo',
+    u'references': {u'internal': u'yes'}}}]
+
+.. _localfile:
+
+Local file with & without fragments
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Parsing references to local files via relative or absolute filepaths work fine, as well as prepending the URI with ``file:``, e.g. ``file:local.schema.json``.
+
+
+RAML File that includes a JSON file under ``schemas``:
+
+.. code-block:: yaml
+
+    #%RAML 0.8
+    title: Sample API Demo - JSON Includes
+    version: v1
+    schemas:
+        - jsonexample: !include local.schema.json
+    baseUri: http://json.example.com
+    /foo:
+      displayName: foo resource
+
+
+The included ``local.schema.json`` file that refers to another JSON file via a relative filepath and a fragment:
+
+.. code-block:: json
+
+    {
+      "$schema":"http://json-schema.org/draft-03/schema",
+      "type": "object",
+      "properties": {
+        "album_type": {
+          "type": "string",
+          "description": "The type of the album: one of 'album', 'single', or 'compilation'."
+        },
+        "artists": {
+            "type": "array",
+            "description": "The artists of the album. Each artist object includes a link in href to more detailed information about the artist.",
+            "items": [{ "$ref": "artist.schema.json#properties" }]
```

Comment:
You're right w/r/t `id` should be a string and a URI if defined ([here's what it says in the draft 4 spec](https://tools.ietf.org/html/draft-zyp-json-schema-04#section-7.2)) - although it's not required.  

Relative `$ref`s [should be resolved](https://tools.ietf.org/html/draft-zyp-json-schema-03#section-5.28) against the URI of the current schema (the `id` of the closest parent).  If a relative `$ref` is defined with no `id` in the schema, then - if I read [this section](https://tools.ietf.org/html/draft-zyp-json-schema-03#section-5.27) correctly, you assume a relative `$ref` is relative to the URI of the parent (which I guess, if loaded locally, then it'd be the current directory).  FWIW the `jsonref` library _looks_ like it will take preference of a defined `id` over a provided `base_uri` parameter, which seems like the correct way given how the RFC for draft 3 reads (I should also write a test case for this).

Re: protocols - the RFCs only state **URIs**, which doesn't necessarily mean that protocol scheme is required (I honestly had to [look this up](https://danielmiessler.com/study/url_vs_uri/) - e.g.  `files.example.com` is a _URI_, `ftp://files.example.com` and `http://files.example.com` are _URLs_).  So the `id` URI does not need to have a protocol attached to it.  I'm not sure how `jsonref` handles undefined schemes (will also make a test case for this).

I do agree that the docs _should_ teach folks how to write good RAML/JSON (it's honestly how I learn best regarding best practices and such).  I do see some RAML-specific questions within the `#ramlfications` freenode channel (partly because `#raml` channel never has any RAML folks in it despite them owning the channel).  I will create an issue at some point today to help improve that aspect.

Not sure if I just went off on a tangent - I honestly spent an hour or so this morning rereading RFCs  and I may have gone done a rabbit hole :)  But this convo has definitely helped me to learn a lot!